src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -855,31 +888,48 @@ LookupCitusTableCacheEntry(Oid relationId)
 		}
 	}
 
-	CitusTableCacheEntry *cacheEntry = hash_search(DistTableCacheHash, hashKey,
-												   HASH_ENTER,
-												   &foundInCache);
+	/*
+	 * We might have some concurrent metadata changes. In order to get the changes,
+	 * we first need to accept the cache invalidation messages.
+	 */
+	AcceptInvalidationMessages();
```

Comment:
@marcocitus Is there a reason why this was moved here, instead of in the `foundInCache` branch?

Reply:
I think Philip's branch did that because invalidation might remove the entry, so it could change the hash. That's no longer the case so we could move it back. It's probably not very consequential for performance, since invalidations eventually need to be processed.